src/plugins/openInApp/index.ts

```diff
@@ -18,46 +18,70 @@
 
 import { definePluginSettings } from "@api/Settings";
 import { Devs } from "@utils/constants";
-import definePlugin, { OptionType, PluginNative } from "@utils/types";
+import definePlugin, { OptionType, PluginNative, SettingsDefinition } from "@utils/types";
 import { showToast, Toasts } from "@webpack/common";
 import type { MouseEvent } from "react";
 
-const ShortUrlMatcher = /^https:\/\/(spotify\.link|s\.team)\/.+$/;
-const SpotifyMatcher = /^https:\/\/open\.spotify\.com\/(track|album|artist|playlist|user|episode)\/(.+)(?:\?.+?)?$/;
-const SteamMatcher = /^https:\/\/(steamcommunity\.com|(?:help|store)\.steampowered\.com)\/.+$/;
-const EpicMatcher = /^https:\/\/store\.epicgames\.com\/(.+)$/;
-const TidalMatcher = /^https:\/\/tidal\.com\/browse\/(track|album|artist|playlist|user|video|mix)\/(.+)(?:\?.+?)?$/;
+type URLReplacementRule = {
+    match: RegExp;
+    replace: (...matches: string[]) => string;
+    description: string;
+    shortlinkMatch?: RegExp;
+    accountViewReplace?: (userId: string) => string;
+};
 
-const settings = definePluginSettings({
+// Do not forget to add protocols to the ALLOWED_PROTOCOLS constant
+const urlReplacementRules: Record<string, URLReplacementRule> = {
```

Comment:
I would use PascalCase / SCREAMING_SNAKE_CASE for constant stuff like this so it's clearer it's not a local variable

```suggestion
const UrlReplacementRules: Record<string, URLReplacementRule> = {
```